x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/autoscaling/MlMemoryAutoscalingDecider.java

```diff
@@ -259,7 +259,7 @@ public MlMemoryAutoscalingCapacity scale(
             .map(result -> {
                 MlMemoryAutoscalingCapacity capacity = ensureScaleDown(
                     result,
-                    MlMemoryAutoscalingCapacity.from(context.currentCapacity()).build()
+                    context.currentCapacity() == null ? null : MlMemoryAutoscalingCapacity.from(context.currentCapacity()).build()
```

Comment:
- `context.currentCapacity()` can be `null`, see [JavaDoc](https://github.com/elastic/elasticsearch/blob/main/x-pack/plugin/autoscaling/src/main/java/org/elasticsearch/xpack/autoscaling/capacity/AutoscalingDeciderContext.java#L24-L27).
- `MlMemoryAutoscalingCapacity.from(null)` gives null pointer exception
- `ensureScaleDown(..., null) = null`